contracts/app/Bridge.sol

```diff
@@ -33,13 +33,14 @@ contract BridgeSource {
     constructor() {}
 
     function withdraw(
+        address sourceTokenAddress,
         BridgeLib.TransferData memory transferData,
         bytes32[] memory proof
     ) public {
         // safemath not needed for solidity 8
         uint256 amountPlusFee = (transferData.amount *
             (10000 + CONTRACT_FEE_BASIS_POINTS)) / 10000;
-        IERC20(transferData.tokenAddress).safeTransferFrom(
+        IERC20(sourceTokenAddress).safeTransferFrom(
```

Comment:
I think evil things can happen if designed like this...

Reply:
Do you mean that we should judge whether the sourceToken is empty and handle it in a different way?

Reply:
Seems the token mapping has to be centralized, can't be decentralized..